components/org.wso2.carbon.identity.verification.onfido.api/org.wso2.carbon.identity.verification.onfido.api.v1/src/main/java/org/wso2/carbon/identity/verification/onfido/api/v1/core/OnfidoIdvService.java

```diff
@@ -0,0 +1,542 @@
+     */
+    private Map<String, String> getIdVConfigPropertyMap(IdVProvider idVProvider) throws OnfidoServerException {
+
+        Map<String, String> configPropertyMap = new HashMap<>();
+        for (IdVConfigProperty idVConfigProperty : idVProvider.getIdVConfigProperties()) {
+            configPropertyMap.put(idVConfigProperty.getName(), idVConfigProperty.getValue());
+        }
+        validateIdVProviderConfigProperties(configPropertyMap);
+        return configPropertyMap;
+    }
+
+    /**
+     * Validates the configuration properties of the Identity Verification Provider.
+     *
+     * @param idVProviderConfigProperties The configuration properties of the Identity Verification Provider.
+     * @throws OnfidoServerException If any required configuration property is missing or invalid.
+     */
+    private void validateIdVProviderConfigProperties(Map<String, String> idVProviderConfigProperties)
+            throws OnfidoServerException {
+
+        if (idVProviderConfigProperties == null || idVProviderConfigProperties.isEmpty() ||
+                StringUtils.isBlank(idVProviderConfigProperties.get(TOKEN)) ||
+                StringUtils.isBlank(idVProviderConfigProperties.get(BASE_URL)) ||
+                StringUtils.isBlank(idVProviderConfigProperties.get(WEBHOOK_TOKEN)) ||
+                StringUtils.isBlank(idVProviderConfigProperties.get(WORKFLOW_ID))) {
+
+            throw new OnfidoServerException(ERROR_IDV_PROVIDER_CONFIG_PROPERTIES_EMPTY.getCode(),
+                    ERROR_IDV_PROVIDER_CONFIG_PROPERTIES_EMPTY.getMessage());
+        }
+    }
+
+    /**
+     * Validates the signature provided in the webhook request against the expected signature.
+     * Implementation logic extracted from https://github.com/onfido/onfido-java/blob/2466de99b6036a8e72186e52bbd5e66e3779223d/src/main/java/com/onfido/WebhookEventVerifier.java#L81
+     *
+     * @param xSHA2Signature              The SHA-2 signature from the Onfido webhook.
+     * @param idVProviderConfigProperties The configuration properties of the Identity Verification Provider.
+     * @param rawRequestBody              The raw verification request payload from Onfido.
+     * @throws OnfidoClientException If the signature validation fails.
+     * @throws OnfidoServerException If a server-side error occurs during validation.
+     */
+    private void validateSignature(String xSHA2Signature, Map<String, String> idVProviderConfigProperties,
+                                   String rawRequestBody) throws OnfidoClientException, OnfidoServerException {
+
+
+        if (StringUtils.isBlank(xSHA2Signature)) {
+            throw new OnfidoClientException(ERROR_SIGNATURE.getCode(), ERROR_SIGNATURE.getMessage());
+        }
+
+        String webhookToken = idVProviderConfigProperties.get(WEBHOOK_TOKEN);
+
+        Mac sha256Hmac;
+        SecretKeySpec secretKey;
+        try {
+            sha256Hmac = Mac.getInstance("HmacSHA256");
```

Comment:
Shall we move this value to a static variable in Constants class?

Reply:
Resolved via https://github.com/wso2-extensions/identity-verification-onfido/pull/12/commits/6f73cff0b813bfcfffbd6cfbe9d730df1f5fe3ed